site/docs/plugins/commands.md

```diff
@@ -5,10 +5,628 @@ next: false
+  // And then you can set the localized ones
+  .localize("pt", "ola", "Dizer olá");
+```
+
+Add as many as you want! The plugin will take care of registering them for you when you call `myCommands.setCommands`.
+
+For convenience, grammY exports a `LanguageCodes` enum-like object that you can use for a more idiomatic approach:
+
+::: code-group
+
+```ts [TypeScript]
+import { LanguageCodes } from "grammy/types";
+
+myCommands.command(
+  "chef",
+  "Steak delivery",
+  (ctx) => ctx.reply("Steak on the plate!"),
+)
+  .localize(
+    LanguageCodes.Spanish,
+    "cocinero",
+    "Bife a domicilio",
+  );
+```
+
+```js [JavaScript]
+const { LanguageCodes } = require("grammy/types");
+
+myCommands.command(
+  "chef",
+  "Steak delivery",
+  (ctx) => ctx.reply("Steak on the plate!"),
+)
+  .localize(
+    LanguageCodes.Spanish,
+    "cocinero",
+    "Bife a domicilio",
+  );
+```
+
+```ts [Deno]
+import { LanguageCodes } from "https://deno.land/x/grammy/types.ts";
+
+myCommands.command(
+  "chef",
+  "Steak delivery",
+  (ctx) => ctx.reply("Steak on the plate!"),
+)
+  .localize(
+    LanguageCodes.Spanish,
+    "cocinero",
+    "Bife a domicilio",
+  );
+```
+
+:::
+
+### Localizing Commands With the Internationalization Plugin
+
+If you are looking to have your localized command names and descriptions bundle inside your `.ftl` files, you could make use of the following idea:
+
+```ts
+function addLocalizations(command: Command) {
+  i18n.locales.forEach((locale) => {
+    command.localize(
+      locale,
+      i18n.t(locale, `${command.name}.command`),
+      i18n.t(locale, `${command.name}.description`),
+    );
+  });
+  return command;
+}
+
+myCommands.commands.forEach(addLocalizations);
+```
+
+## Finding the Nearest Command
+
+Even though Telegram is capable of auto completing the registered commands, sometimes users do type them manually and, in some cases, happen to make mistakes.
+The commands plugin helps you deal with that by allowing you to suggest a command that might be what the user wanted in the first place.
+It is compatible with custom prefixes, so you don't have to worry about that, and its usage is quite straightforward:
+
+::: code-group
+
+```ts [TypeScript]
+// Use the flavor to create a custom context
+type MyContext = Context & CommandsFlavor;
+
+// Use the new context to instantiate your bot
+const bot = new Bot<MyContext>("token");
+const myCommands = new CommandGroup<MyContext>();
+
+// ... Register the commands
+
+bot
+  // Check if there is a command
+  .filter(commandNotFound(myCommands))
+  // If so, that means it wasn't handled by any of our commands.
+  .use(async (ctx) => {
+    // We found a potential match
+    if (ctx.commandSuggestion) {
+      await ctx.reply(
+        `Hmm... I don't know that command. Did you mean ${ctx.commandSuggestion}?`,
+      );
+    }
+
+    // Nothing seems to come close to what the user typed
+    await ctx.reply("Oops... I don't know that command :/");
+  });
+```
```

Comment:
This code throws an error.

Error:
```
Error in middleware while handling update 141326753 TypeError: ctx.getNearestCommand is not a function
    at https://deno.land/x/grammy_commands@v0.10.0/command-group.ts:398:12
    at https://deno.land/x/grammy@v1.31.0/composer.ts:863:20
    at https://deno.land/x/grammy@v1.31.0/composer.ts:784:38
    at https://deno.land/x/grammy@v1.31.0/composer.ts:141:39
    at https://deno.land/x/grammy@v1.31.0/composer.ts:152:19
    at pass (https://deno.land/x/grammy@v1.31.0/composer.ts:157:12)
    at https://deno.land/x/grammy@v1.31.0/composer.ts:141:39
    at https://deno.land/x/grammy@v1.31.0/composer.ts:786:43
    at eventLoopTick (ext:core/01_core.js:175:7)
    at async https://deno.land/x/grammy@v1.31.0/composer.ts:152:13
No error handler was set!
Set your own error handler with `bot.catch = ...`
Stopping bot
error: Uncaught (in promise) BotError: TypeError in middleware: ctx.getNearestCommand is not a function 
```

The code:
```ts
import { Bot } from "https://deno.land/x/grammy@v1.31.0/mod.ts";
import {
  CommandGroup,
  commandNotFound,
} from "https://deno.land/x/grammy_commands@v0.10.0/mod.ts";

const botToken = Deno.env.get("BOT_TOKEN");
if (!botToken) throw new Error("Missing bot token!");

const bot = new Bot(botToken);

const myCommands = new CommandGroup();

myCommands
  .command("hello", "Say hello", async (ctx) => {
    await ctx.reply("Hello, little user!");
  });

bot.use(myCommands);

bot
  // Check if there is a command
  .filter(commandNotFound(myCommands))
  // If so, that means it wasn't handled by any of our commands
  .use(async (ctx) => {
    // We found a potential match
    if (ctx.commandSuggestion) {
      await ctx.reply(
        `Hmm... I don't know that command. Did you mean ${ctx.commandSuggestion}?`,
      );
    }

    // Nothing seems to come close to what the user typed
    await ctx.reply("Oops... I don't know that command :/");
  });

bot.start({
  onStart() {
    console.info("Ready.");
  },
});
```

Reply:
Yes. It's missing the plug-in installation

Reply:
> Yes. It's missing the plug-in installation

Which is included in this doc example, in the `bot.ts` file. But missing in single file snippet @quadratz send. Add `bot.use(commands());`and you should be good